### PR TITLE
Feature/realm

### PR DESCRIPTION
**Pull request by cgx**
This branch adds the possibility to use the RADIUS realm to filter which authentication source(s) to use to retrieve the normal VLAN.

Update:
## Description

This branch manage the realm in packetfence:
Add a file in radius configuration to set the realm and the options
Use or no the stripped user to match authentication source
Add a portal filter to match on the realm
## Impacts

This will change the way to deal with the authentication sources and the realm hasn´t been defined then we will receive username attribute like that (user@domain.ca or DOMAIN\User) in packetfence.
## Delete branch after merge

YES

### CODE REVIEW

**Comment by fdurand** [comment]
Review please

**Comment by fdurand** [comment]
Where do you find this code ?

File: html/pfappserver/lib/pfappserver/Controller/Realm.pm [hunk 1]
```diff
@@ -0,0 +1,111 @@
+package pfappserver::Controller::Configuration::Realm;
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Controller/Realm.pm]: Should be pfappserver::Controller::Config::Realm

File: html/pfappserver/root/configuration/realm/index.tt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+      <div class="modal fade hide" id="modalItem">
```
Comment (jrouzierinverse) [inline html/pfappserver/root/configuration/realm/index.tt]: rename to html/pfappserver/root/configuration/realm/index.tt html/pfappserver/root/config/realm/index.tt

File: html/pfappserver/root/configuration/realm/list.tt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+      <table id="items" class="table items" data-list-uri="[% c.uri_for(c.controller.action_for('list')) %]">
```
Comment (jrouzierinverse) [inline html/pfappserver/root/configuration/realm/list.tt]: rename to html/pfappserver/root/config/realm/list.tt html/pfappserver/root/config/realm/list.tt

File: html/pfappserver/root/configuration/realm/view.tt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+[% UNLESS action_uri %]
```
Comment (jrouzierinverse) [inline html/pfappserver/root/configuration/realm/view.tt]: rename to html/pfappserver/root/config/realm/view.tt html/pfappserver/root/config/realm/view.tt

File: lib/pf/radius.pm [hunk 5]
```diff
@@ -177,9 +177,11 @@ sub authorize {
 
     # Check if a floating just plugged in
     $this->_handleAccessFloatingDevices($switch, $mac, $port);
+    # Extract the realm
+    my $realm = $radius_request->{'Realm'};
```
Comment (jrouzierinverse) [inline lib/pf/radius.pm]: Already defined previously

File: html/pfappserver/lib/pfappserver/Controller/Realm.pm [hunk 6]
```diff
@@ -0,0 +1,111 @@
+package pfappserver::Controller::Config::Realm;
+
+=head1 NAME
+
+pfappserver::Controller::Configuration::Realm - Catalyst Controller
+
+=head1 DESCRIPTION
+
+Controller for firewall sso  management.
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Controller/Realm.pm]: Rewrite description.

File: html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/Htpasswd.pm [hunk 7]
```diff
@@ -21,6 +21,16 @@ has_field 'path' =>
...
   );
+has_field 'stripped_user_name' =>
+  (
+   type            => 'Toggle',
+   checkbox_value  => 'yes',
+   unchecked_value => 'no',
+   default         => 'yes',
+   label           => 'Use stripped username ',
+   tags => { after_element => \&help,
+             help => 'Use stripped username returned by radius to test the following rules.' },
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/Htpasswd.pm]: _RADIUS_ must be all in caps.

File: html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/Kerberos.pm [hunk 8]
```diff
@@ -28,6 +28,16 @@ has_field 'realm' =>
...
   );
+has_field 'stripped_user_name' =>
+  (
+   type            => 'Toggle',
+   checkbox_value  => 'yes',
+   unchecked_value => 'no',
+   default         => 'yes',
+   label           => 'Use stripped username ',
+   tags => { after_element => \&help,
+             help => 'Use stripped username returned by radius to test the following rules.' },
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/Kerberos.pm]: _RADIUS_ must be all in caps.

File: lib/pf/ConfigStore/Realm.pm [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+use warnings;
+use Moo;
+use pf::file_paths;
+extends 'pf::ConfigStore';
+
+sub configFile { $pf::file_paths::realm_config_file };
+
+=head2 cleanupAfterRead
+
+Clean up switch data
```
Comment (cgx) [inline lib/pf/ConfigStore/Realm.pm]: Switch data?

**Comment by fdurand** [comment]
Can it be merge ?

**Comment by fdurand** [comment]
Branch has been rebased for i hope to be merge

**Pull request merged**